src/ess/reduce/correction.py

```diff
@@ -0,0 +1,193 @@
+    -------
+    :
+        `detector` normalized by a monitor.
+
+    See also
+    --------
+    normalize_by_monitor_histogram:
+        Normalize by a monitor histogram without integration.
+    """
+    clipped = _clip_monitor_to_detector_range(monitor=monitor, detector=detector)
+    coord = clipped.coords[clipped.dim]
+    norm = (clipped * (coord[1:] - coord[:-1])).data.sum()
+    norm = broadcast_uncertainties(
+        norm, prototype=detector, mode=uncertainty_broadcast_mode
+    )
+    return detector / norm
+
+
+def _clip_monitor_to_detector_range(
+    *, monitor: sc.DataArray, detector: sc.DataArray
+) -> sc.DataArray:
+    dim = monitor.dim
+    if not monitor.coords.is_edges(dim):
+        raise sc.CoordError(
+            f"Monitor coordinate '{dim}' must be bin-edges to integrate the monitor."
+        )
+
+    # Prefer a bin coord over an event coord because this makes the behavior for binned
+    # and histogrammed data consistent. If we used an event coord, we might allow a
+    # monitor range that is less than the detector bins which is fine for the vents,
+    # but would be wrong if the detector was subsequently histogrammed.
+    if dim in detector.coords:
+        det_coord = detector.coords[dim]
+
+        # Mask zero-count bins, which are an artifact from the rectangular 2-D binning.
+        # The wavelength of those bins must be excluded when determining the range.
+        if detector.bins is None:
+            mask = detector.data == sc.scalar(0.0, unit=detector.unit)
+        else:
+            mask = detector.data.bins.size() == sc.scalar(0.0, unit=None)
+        lo = (
+            sc.DataArray(det_coord[dim, :-1], masks={'zero_counts': mask}).nanmin().data
+        )
+        hi = sc.DataArray(det_coord[dim, 1:], masks={'zero_counts': mask}).nanmax().data
+
+    elif dim in detector.bins.coords:
+        det_coord = detector.bins.coords[dim]
+        # No need to mask here because we have the exact event coordinate values.
+        lo = det_coord.nanmin()
+        hi = det_coord.nanmax()
+
+    else:
+        raise sc.CoordError(
+            f"Missing '{dim}' coordinate in detector for monitor normalization."
+        )
+
+    if monitor.coords[dim].min() > lo or monitor.coords[dim].max() < hi:
+        raise ValueError(
+            f"Cannot normalize by monitor: The {dim} range of the monitor "
+            f"({monitor.coords[dim].min().value} to {monitor.coords[dim].max().value}) "
+            f"is smaller than the range of the detector ({lo.value} to {hi.value})."
+        )
+
+    if detector.bins is None:
+        # If we didn't rebin to the detector coord here, then, for a finer monitor
+        # binning than detector, the lookup table would extract one monitor value for
+        # each detector bin and ignore other values lying in the same detector bin.
+        # But integration would pick up all monitor bins.
+        return monitor.rebin({dim: det_coord})
+    return monitor[dim, lo:hi]
```

Comment:
Changes look good but can you briefly outline which code was simply copied over and which code is new? Thanks :-)

Reply:
Nothing is copied without change. But `normalize_by_monitor_integrated` is unchanged apart from extracting `_clip_monitor_to_detector_range`. However, the latter has changed as I described in my initial comment.

`normalize_by_monitor_histogram` now also clips the data. And the actual normalisation has changed. It used to be https://github.com/scipp/essdiffraction/blob/c9c29dd6155cdb0f9fd201e6c9d6923a754b1e37/src/ess/powder/correction.py#L57-L67, i.e., only `det / mon` without extra scaling factors.